File: go-client-sdk/pkg/models/operations/getcompaniescompanyidconnectionsconnectioniddataaccounttransactionsaccounttransactionid.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetCompaniesCompanyIDConnectionsConnectionIDDataAccountTransactionsAccountTransactionIDPathParams struct {
	AccountTransactionID string `pathParam:"style=simple,explode=false,name=accountTransactionId"`
	CompanyID            string `pathParam:"style=simple,explode=false,name=companyId"`
	ConnectionID         string `pathParam:"style=simple,explode=false,name=connectionId"`
}

type GetCompaniesCompanyIDConnectionsConnectionIDDataAccountTransactionsAccountTransactionIDSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDConnectionsConnectionIDDataAccountTransactionsAccountTransactionIDRequest struct {
	PathParams GetCompaniesCompanyIDConnectionsConnectionIDDataAccountTransactionsAccountTransactionIDPathParams
	Security   GetCompaniesCompanyIDConnectionsConnectionIDDataAccountTransactionsAccountTransactionIDSecurity
}

type GetCompaniesCompanyIDConnectionsConnectionIDDataAccountTransactionsAccountTransactionIDResponse struct {
	CodatDataContractsDatasetsAccountTransaction *shared.CodatDataContractsDatasetsAccountTransaction
	ContentType                                  string
	StatusCode                                   int
}

File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyidconnectionsconnectioniddataaccounttransactionsaccounttransactionid.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class GetCompaniesCompanyIdConnectionsConnectionIdDataAccountTransactionsAccountTransactionIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=accountTransactionId" })
  accountTransactionId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=connectionId" })
  connectionId: string;
}

export class GetCompaniesCompanyIdConnectionsConnectionIdDataAccountTransactionsAccountTransactionIdSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetCompaniesCompanyIdConnectionsConnectionIdDataAccountTransactionsAccountTransactionIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdConnectionsConnectionIdDataAccountTransactionsAccountTransactionIdPathParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdConnectionsConnectionIdDataAccountTransactionsAccountTransactionIdSecurity;
}

export class GetCompaniesCompanyIdConnectionsConnectionIdDataAccountTransactionsAccountTransactionIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatDataContractsDatasetsAccountTransaction?: shared.CodatDataContractsDatasetsAccountTransaction;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}